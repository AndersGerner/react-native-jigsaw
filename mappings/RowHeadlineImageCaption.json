[{"name":"Row Single Line Headline Caption","tag":"RowHeadlineImageCaption","description":"A row with left aligned headline text and right aligned caption text","category":"deprecated","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_SingleLine_HeadlineCaption.png","supports_list_render":true,"props":{"title":{"label":"Title","description":"Headline text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"caption":{"label":"Caption","description":"Caption text to display","type":"string","value":"$100","editable":true}},"layout":{}},{"name":"Row Single Line Headline Image Caption","tag":"RowHeadlineImageCaption","description":"A row with left aligned image and headline text and right aligned caption text","category":"deprecated","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_SingleLine_HeadlineImageCaption.png","supports_list_render":true,"props":{"title":{"label":"Title","description":"Headline text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"caption":{"label":"Caption","description":"Caption text to display","type":"string","value":"$100","editable":true},"image":{"label":"Image","description":"Image to display","type":"remoteImage","value":null,"editable":true}},"layout":{}},{"name":"Row Double Line Headline Caption","tag":"RowHeadlineImageCaption","description":"A row with left aligned headline text and subtitle text and right aligned caption text","category":"deprecated","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_DoubleLine_HeadlineCaption.png","supports_list_render":true,"props":{"title":{"label":"Title","description":"Headline text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"subtitle":{"label":"Subtitle","description":"Subtitle text to display","type":"string","value":"San Diego","editable":true},"caption":{"label":"Caption","description":"Caption text to display","type":"string","value":"$100","editable":true}},"layout":{}},{"name":"Row Double Line Headline Image Caption","tag":"RowHeadlineImageCaption","description":"A row with left aligned image, headline text, and subtitle text, and right aligned caption text","category":"deprecated","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_DoubleLine_HeadlineCaption.png","supports_list_render":true,"props":{"title":{"label":"Title","description":"Headline text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"subtitle":{"label":"Subtitle","description":"Subtitle text to display","type":"string","value":"San Diego","editable":true},"caption":{"label":"Caption","description":"Caption text to display","type":"string","value":"$100","editable":true},"image":{"label":"Image","description":"Image to display","type":"remoteImage","value":null,"editable":true}},"layout":{}}]