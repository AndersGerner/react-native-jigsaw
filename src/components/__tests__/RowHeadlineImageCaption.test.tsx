import * as React from "react";
import renderer from "react-test-renderer";
import RowHeadlineImageCaption from "../RowHeadlineImageCaption";

it("renders row headline image caption", () => {
  const tree = renderer
    .create(
      <RowHeadlineImageCaption image={require("../../assets/icon.png")} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
