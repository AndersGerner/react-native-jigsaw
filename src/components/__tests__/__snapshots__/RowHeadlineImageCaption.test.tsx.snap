// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders row headline image caption 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "padding": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 24,
          "marginRight": 12,
          "width": 24,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(18, 20, 44, 1)",
            },
          ]
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "rgba(18, 20, 44, 1)",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 16,
        "marginLeft": 16,
      }
    }
  />
</View>
`;
